src/store/setToolMode.js

```diff
@@ -125,17 +216,104 @@ function setToolModeForElement (mode, changeEvent, element, toolName, options) {
 /**
  * A helper/quick way to set a tool's mode for all canvases
  *
- * @param {*} mode
- * @param {*} changeEvent
- * @param {*} toolName
- * @param {*} options
+ * @param {string} mode
+ * @param {string} changeEvent
+ * @param {string} toolName
+ * @param {(object|number)} options
+ * @param {boolean} isTouchActive
  */
-function setToolMode (mode, changeEvent, toolName, options) {
+function setToolMode (mode, changeEvent, toolName, options, isTouchActive) {
   state.enabledElements.forEach((element) => {
-    setToolModeForElement(mode, changeEvent, element, toolName, options);
+    setToolModeForElement(
+      mode,
+      changeEvent,
+      element,
+      toolName,
+      options,
+      isTouchActive
+    );
   });
 }
 
+/**
+ *
+ *
+ * @param {*} element
+ * @param {*} options
+ * @param {*} isTouchActive
+ * @private
+ */
+function _setActiveToolWithMatchingMouseButtonMaskToPassive (
+  tool,
+  element,
+  options
+) {
+  const mouseButtonMask =
+    (typeof options === 'number' ? options : options.mouseButtonMask) ||
```

Comment:
Again, maybe we could do something a bit neater with default options to make it so checks like this don't need to happen? Is there a particular reason that options should be just a number?